borsh/src/ser/mod.rs

```diff
@@ -201,6 +204,38 @@ impl BorshSerialize for String {
     }
 }
 
+#[cfg(feature = "bigdecimal")]
+impl BorshSerialize for bigdecimal::BigDecimal {
+    #[inline]
+    fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
+        let (bigint, exponent) = self.as_bigint_and_exponent();
+        bigint.serialize(writer)?;
+        exponent.serialize(writer)
+    }
+}
+
+#[cfg(feature = "bigdecimal")]
+impl BorshSerialize for bigdecimal::num_bigint::BigInt {
```

Comment:
Noting that if providing `BigInt`, `BigUInt` is also a common type used for various reasons, so possibly worth adding that. Can be done in separate PR though (and we can just open an issue if not done here)